### PR TITLE
dcp-628 : New project from geo

### PR DESCRIPTION
**Pull request by aaclan-ebi**
dcp-628

### CODE REVIEW

File: broker/import_geo/routes.py [hunk 1]
```diff
@@ -1,47 +1,88 @@
...
-                             attachment_filename=filename)
+def _import_project_from_workbook(workbook):
+    importer = XlsImporter(app.ingest_api)
+    token = request.headers.get('Authorization')
+    project_uuid, errors = importer.import_project_from_workbook(workbook, token)
+
+    if errors:
+        error_details = [e.get('details') for e in errors if e.get('details')]
+        error_messages = ' ,'.join(error_details)
+        raise ImportProjectWorkbookError(f'There were errors in importing the project: {error_messages}.')
```
Comment (yusra-haider) [inline broker/import_geo/routes.py:71]: nit: I think these could be combined to something like this:

```
' ,'.join([e.get('details') for e in errors if e.get('details')])

```

**Review by yusra-haider — DISMISSED** [review]

**Comment by yusra-haider** [comment]
LGTM 👍

File: broker/import_geo/routes.py [hunk 2]
```diff
@@ -1,47 +1,88 @@
...
+
+
+@import_geo_bp.errorhandler(ImportGeoHttpError)
+def handle_import_geo_http_error(e: ImportGeoHttpError):
+    return response_json(e.status_code, {'message': e.message})
+
+
+def _generate_geo_workbook(geo_accession: str):
+    if not _is_valid_geo_accession(geo_accession):
+        raise InvalidGeoAccession('The given geo accession is invalid.')
```
Comment (amnonkhen) [inline broker/import_geo/routes.py]: I think the offending string would be useful here.

Comment (aaclan-ebi) [reply broker/import_geo/routes.py]: Applied! thanks!

**Review by jacobwindsor — DISMISSED** [review]

**Review by yusra-haider — DISMISSED** [review]

**Review by yusra-haider — APPROVED** [review]

**Review by jacobwindsor — APPROVED** [review]

**Review by prabh-t — APPROVED** [review]
Looks good to me. Please consider renaming the `import_geo` package.

**Comment by yusra-haider** [comment]
merged to dev

**Review by amnonkhen — COMMENTED** [review]
LGTM

**Pull request merged**